### DISCUSSION TITLE
No licence key found! - Failed to read GeoIP database and Essentials

### DISCUSSION THREAD

**Discussion by Skjolberg** [EssentialsX/Essentials#4283] [Q&A] [answered]
First of all, sorry for my English since I do not have the native language.

I have updated to 1.17 seeing that there is a version of Essentials for 1.17 (although beta) and it is giving me an error in console that I put it in an image so that it is understood.

Geo-IP: https://i.gyazo.com/c495898f69f92793d5cf3ada970475cb.png
Essentials: https://i.gyazo.com/ba10ba67ffec995f0f5fcce2a0a53acd.png

Thanks for the plugin, it is incredible what you do.

**Comment by pop4959** [comment, marked as answer]
For the first error, please follow directions on the website as provided. GeoIP (for a while now) requires that you create a MaxMind account to request a license key to download and access their database.

For the second error, please report as an issue and fill out the bug template.

As for the "limited API functionality", you can fix that by using Paper instead of Spigot.

**Reply by Skjolberg** [reply]
> Para el primer error, siga las instrucciones proporcionadas en el sitio web. GeoIP (desde hace un tiempo) requiere que cree una cuenta MaxMind para solicitar una clave de licencia para descargar y acceder a su base de datos.
> 
> Para el segundo error, notifíquelo como un problema y complete la plantilla de error.
> 
> En cuanto a la "funcionalidad API limitada", puede solucionarlo utilizando Paper en lugar de Spigot.

Essentials not good for spigot? Thx

**Reply by pop4959** [reply]
Essentials can be used with Spigot but it will just be missing a few features and some commands.

**Reply by Skjolberg** [reply]
> Essentials se puede usar con Spigot, pero solo le faltarán algunas funciones y algunos comandos.

When the version of Essentials is stable ... will it work for Spigot 1.17 perfectly? thx

**Reply by mdcfe** [reply]
> > Essentials se puede usar con Spigot, pero solo le faltarán algunas funciones y algunos comandos.
> 
> When the version of Essentials is stable ... will it work for Spigot 1.17 perfectly? thx

Spigot is missing certain features that EssentialsX and benefits from, and has lacked these features for years. There is nothing EssentialsX can do about Spigot's lack of features without adding a bunch of hacks around server code. If you want features like `/recipe`, `/stonecutter` and faster async teleports, you'll need to use Paper.
